homeassistant/components/oncue/entity.py

```diff
@@ -53,3 +53,19 @@ def _oncue_value(self) -> str:
         device: OncueDevice = self.coordinator.data[self._device_id]
         sensor: OncueSensor = device.sensors[self.entity_description.key]
         return sensor.value
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        if self._oncue_value == VALUE_UNAVAILABLE:
+            return False
+        # If the cloud is reporting that the generator is not connected
+        # this also indicates the data is not available.
+        # The battery voltage sensor reports 0.0 rather than -- hence the purpose of this check.
+        # However, the binary sensor that tracks the connection should not go unavailable.
+        if self.entity_description.key != CONNECTION_ESTABLISHED_KEY:
+            device: OncueDevice = self.coordinator.data[self._device_id]
+            conn_established: OncueSensor = device.sensors[CONNECTION_ESTABLISHED_KEY]
+            if conn_established is not None and conn_established.value == "false":
+                return False
```

Comment:
In a future PR it might be nice to subclass the coordinator and make this a property that gets set during updates so we don't have to do all the dict lookups for each entity when state is written.

Reply:
Agree